server/py/framework/db/sqldb/db.py

```diff
@@ -5580,48 +5655,103 @@ def _transform_model_endpoint_model_to_schema(
+            model_endpoint_record, model_endpoint_full_dict
+        )
 
         model_endpoint_full_dict = (
             mlrun.common.formatters.ModelEndpointFormat.format_obj(
                 model_endpoint_full_dict, format_
             )
         )
+
         model_endpoint_resp = mlrun.common.schemas.ModelEndpoint.from_flat_dict(
             model_endpoint_full_dict
         )
-
+        model_endpoint_full_dict["_model_id"] = None
         return model_endpoint_resp
 
-    @staticmethod
     def _fill_model_endpoint_with_function_data(
+        self,
         model_endpoint_record: ModelEndpoint,
         model_endpoint_full_dict: dict,
         latest: bool,
     ) -> dict:
         if model_endpoint_record.function and latest:
+            model_endpoint_full_dict[ModelEndpointSchema.FUNCTION_NAME] = (
+                model_endpoint_record.function.name
+            )
+            function_tag_list = model_endpoint_record.function.tags
+            model_endpoint_full_dict[ModelEndpointSchema.FUNCTION_TAGS] = (
+                [tag.name for tag in function_tag_list] if function_tag_list else []
+            )
+            model_endpoint_full_dict[ModelEndpointSchema.FUNCTION_TAG] = (
+                self._get_function_tag(function_tag_list)
+            )
             model_endpoint_full_dict[ModelEndpointSchema.STATE] = (
                 model_endpoint_record.function.state
             )
             model_endpoint_full_dict[ModelEndpointSchema.FUNCTION_URI] = (
                 generate_object_uri(
-                    project=model_endpoint_record.project,
-                    name=model_endpoint_record.function_name,
+                    project=model_endpoint_record.function.project,
+                    name=model_endpoint_record.function.name,
                     hash_key=model_endpoint_record.function.uid,
                 )
             )
+
         else:
+            model_endpoint_full_dict[ModelEndpointSchema.FUNCTION_NAME] = ""
+            model_endpoint_full_dict[ModelEndpointSchema.FUNCTION_TAG] = ""
+            model_endpoint_full_dict[ModelEndpointSchema.FUNCTION_TAGS] = []
             model_endpoint_full_dict[ModelEndpointSchema.STATE] = "unknown"
             model_endpoint_full_dict[ModelEndpointSchema.FUNCTION_URI] = None
         return model_endpoint_full_dict
 
+    def _get_function_tag(self, function_tag_list):
+        latest = False
+        for tag in function_tag_list:
+            if tag.name == mlrun.common.constants.RESERVED_TAG_NAME_LATEST:
+                latest = True
+            else:
+                return tag.name
+        if latest:
+            return mlrun.common.constants.RESERVED_TAG_NAME_LATEST
+        return ""
+
+    @staticmethod
+    def _fill_model_endpoint_with_model_data(
+        model_endpoint_record: ModelEndpoint, model_endpoint_full_dict: dict
+    ) -> dict:
+        model = model_endpoint_record.model
+        if model:
```

Comment:
just some fun python 3.9 feature
```suggestion
        if model := model_endpoint_record.model:
```